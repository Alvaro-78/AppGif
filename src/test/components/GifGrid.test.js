import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('test GifGrid component', () => {
    // Mock data
    useFetchGifs.mockReturnValue({
        data: [],
        loading: true,
    });

    const category = 'random';
    const wrapper = shallow(<GifGrid category={category} />);

    // tes component
    it('should render component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // test our custom Hook return mock data
    it('should show items when images are loaded useFetchGifs', () => {
        const gifs = [
            {
                id: 'abc',
                url: 'https://localhost/loquesea/cosa.jpg',
                title: 'Cualquiercosa',
            },
            {
                id: 'def',
                url: 'https://localhost/loquesea/otracosa.jpg',
                title: 'Cualquier otra cosa',
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
