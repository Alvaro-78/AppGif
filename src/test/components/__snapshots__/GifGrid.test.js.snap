// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifGrid component should render component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeI"
  >
    random
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`test GifGrid component should show items when images are loaded useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeI"
  >
    random
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquiercosa"
      url="https://localhost/loquesea/cosa.jpg"
    />
    <GifGridItem
      id="def"
      key="def"
      title="Cualquier otra cosa"
      url="https://localhost/loquesea/otracosa.jpg"
    />
  </div>
</Fragment>
`;
